File: C/11_Brute_Force/BOJ_17089_세친구.c
#include <stdio.h>
#define math_min(a, b) a < b ? a : b
main() {
	int n, m;
	scanf("%d %d", &n, &m);
	int* cnt = (int*)malloc(sizeof(int) * (n + 1));
	int** friended = (int**)malloc(sizeof(int*) * (n + 1));
	for (int i = 1; i <= n; i++) {
		friended[i] = (int*)malloc(sizeof(int) * (n + 1));
		for (int j = 1; j <= n; j++) {
			friended[i][j] = 0;
		}
		cnt[i] = -2;
	}
	while (m--) {
		int x, y;
		scanf("%d %d", &x, &y);
		friended[x][y] = friended[y][x] = 1;
		cnt[x]++;
		cnt[y]++;
	}
	int min = 2147483647;
	for (int i = 1; i <= n - 2; i++) {
		for (int j = i + 1; j <= n - 1; j++) {
			if (friended[i][j]) {
				for (int k = j + 1; k <= n; k++) {
					if (friended[j][k] && friended[k][i]) {
						int sum = cnt[i] + cnt[j] + cnt[k];
						min = math_min(min, sum);
					}
				}
			}
		}
	}
	if (min == 2147483647) {
		printf("-1");
	}
	else {
		printf("%d", min);
	}
}

File: JAVA/11_Brute_Force/BOJ_17089_세친구.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        int[] cnt = new int[n + 1];
        Arrays.fill(cnt, -2);
        boolean[][] friended = new boolean[n + 1][n + 1];
        while (m-- > 0) {
            st = new StringTokenizer(br.readLine(), " ");
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());
            friended[x][y] = friended[y][x] = true;
            cnt[x]++;
            cnt[y]++;
        }
        int min = Integer.MAX_VALUE;
        for (int i = 1; i <= n - 2; i++) {
            for (int j = i + 1; j <= n - 1; j++) {
                if (friended[i][j]) {
                    for (int k = j + 1; k <= n; k++) {
                        if (friended[j][k] && friended[k][i]) {
                            int sum = cnt[i] + cnt[j] + cnt[k];
                            min = Math.min(min, sum);
                        }
                    }
                }
            }
        }
        if (min == Integer.MAX_VALUE) {
            System.out.println(-1);
        } else {
            System.out.println(min);
        }
    }
}